Source/WebCore/dom/Element.cpp

```diff
@@ -1975,6 +1976,43 @@ const AtomString& Element::getAttributeNS(const AtomString& namespaceURI, const
     return getAttribute(QualifiedName(nullAtom(), localName, namespaceURI));
 }
 
+static ExceptionOr<String> trustedTypesCompliantAttributeValue(const String& attributeType, const TrustedTypeOrString& value, Element* element, String sink)
+{
+    auto stringValueHolder = WTF::switchOn(value,
+        [&](const String& str) -> ExceptionOr<String> {
+            if (attributeType.isNull())
+                return String(str);
+            return trustedTypeCompliantString(stringToTrustedType(attributeType), *(element->document().scriptExecutionContext()), str, sink);
+        },
+        [](const RefPtr<TrustedHTML>& trustedHTML) -> ExceptionOr<String> {
+            return trustedHTML->toString();
+        },
+        [](const RefPtr<TrustedScript>& trustedScript) -> ExceptionOr<String> {
+            return trustedScript->toString();
+        },
+        [](const RefPtr<TrustedScriptURL>& trustedScriptURL) -> ExceptionOr<String> {
+            return trustedScriptURL->toString();
+        }
```

Comment:
This switch is wrong, if an attribute expects TrustedHTML it shouldn't allow you to pass TrustedScript for example.

You need to do something like below (obviously follow code standards)

```c++
if (attributeType == TrustedXXX) return trustedXXX->toString()
else return Exception { TypeError }
```